azurerm/internal/services/storage/client/client.go

```diff
@@ -24,15 +24,17 @@ type Client struct {
 	FileSystemsClient        *filesystems.Client
 	ManagementPoliciesClient storage.ManagementPoliciesClient
 	BlobServicesClient       storage.BlobServicesClient
-	environment              az.Environment
+
+	environment az.Environment
+	storageAuth *autorest.Authorizer
```

Comment:
Given this is not for all storage as some don't support it atm, and just storage ad, auth might it make more sense to be
```suggestion
	storageAdAuth *autorest.Authorizer
```

Reply:
👍